### PR TITLE
Add check for tied parameters

### PR DESCRIPTION
**Pull request by SunMarc**
# What does this PR do 

This is the fix in the transformers library of this [PR](https://github.com/huggingface/accelerate/pull/1529). 

It will fix the case where a user uses their own device map (in the `from_pretrained` method) but forget that parameters that are tied together should be on the same device. We return an error showing which parameters should be on the same device.

### CODE REVIEW

**Comment by SunMarc** [comment]
The test are failing because I used a function that i have added recently in accelerate.utils. Should we use the main for the tests @sgugger ?

**Review by sgugger — COMMENTED** [review]
Thanks for your PR!
You will need to protect the import a bit better as there are many checks (and users!) that do not use the latest of Accelerate, and we want to not break everything for them ;-)
You can put the import in a try except and have the function be `None` when it's not in the version of Accelerate used.

File: src/transformers/modeling_utils.py [hunk 1]
```diff
@@ -2824,6 +2828,13 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
                     )
                 del device_map_without_lm_head
 
+        if device_map is not None:
```
[the remark this answers is not in the document]
Comment (SunMarc) [reply src/transformers/modeling_utils.py:2831]: Yes it's better. I was just considering that we should test the device_map even it is generated by infer_auto_device_map but i guess we already have some tests in place for that.

**Review by sgugger — APPROVED** [review]
LGTM, thanks for iterating!

**Pull request merged**